src/hssm/param/params.py

```diff
@@ -0,0 +1,353 @@
+    p_outlier: float | dict | bmb.Prior | None,
+) -> dict[str, UserParam]:
+    """Collect and convert parameters to UserParam objects.
+
+    Parameters
+    ----------
+    model
+        The HSSM model.
+    include
+        A list of dictionaries or UserParam objects specifying the parameters.
+    kwargs
+        Keyword arguments specifying the parameters.
+    p_outlier
+        The prior specification for the outlier probability.
+
+    Returns
+    -------
+    dict[str, UserParam]
+        A dictionary with the parameter names as keys and UserParam objects as values.
+    """
+    user_params: dict[str, UserParam] = {}
+
+    # Process include
+    for param in include:
+        user_param = UserParam.from_dict(param) if isinstance(param, dict) else param
+        if user_param.name is None:
+            raise ValueError("Parameter name must be specified.")
+        if user_param.name not in model.list_params:
+            raise ValueError(f"Parameter {user_param.name} not found in list_params.")
+        if user_param.name == "p_outlier":
+            raise ValueError(
+                "Please do not specify `p_outlier` in `include`. "
+                "Please specify it with `p_outlier` instead."
+            )
+        user_params[user_param.name] = user_param
+
+    # Process kwargs
+    # If any of the keys is found in `list_params` it is a parameter specification
+    # We add the parameter specification to `user_params` and remove it from
+    # `kwargs`
+    for param_name in model.list_params:
+        if param_name in kwargs:
+            if param_name in user_params:
+                raise ValueError(
+                    f"Parameter {param_name} specified in both `include` and `kwargs`."
+                )
+            user_params[param_name] = UserParam.from_kwargs(
+                param_name,
+                # Use pop to remove the key from kwargs
+                kwargs.pop(param_name),
+            )
+
+    # Process p_outliers the same way.
+    if model.has_lapse:
+        user_params["p_outlier"] = UserParam.from_kwargs("p_outlier", p_outlier)
+
+    return user_params
+
+
+def make_params(model: HSSM, user_params: dict[str, UserParam]) -> dict[str, Param]:
+    """Make parameters from a dict of UserParams.
+
+    Parameters
+    ----------
+    model
+        The HSSM model.
+    user_params
+        A dictionary with the parameter names as keys and UserParam objects as values.
+
+    Returns
+    -------
+    dict[str, Param]
+        A dictionary with the parameter names as keys and Param objects as values.
+    """
+    params = {}
+    is_ddm = (
+        model.model_name in ["ddm", "ddm_sdv", "full_ddm"]
+        and model.loglik_kind != "approx_differentiable"
+    )
+
+    for name in model.list_params:
+        if name in user_params:
+            param: Param = make_param_from_user_param(model, name, user_params[name])
+        else:
+            param = make_param_from_defaults(model, name)
```

Comment:
Type hint?

Reply:
It's actually implicitly the same type as the `Param` in the other branch. I had to have explicit type hint for the other branch because that's mypy will use for this branch as well, which causes a mypy error